File: JavaScript/78. Subsets.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsets = function(nums) {
    var res = [];
    var temp = [];
    helper(0,nums,temp,res);
    return res;
};
var helper = function(i,nums,temp,res){
    if(i==nums.length){
        res.push([...temp]);
        return;
    }
    temp.push(nums[i]);
    helper(i+1,nums,temp,res);
    temp.pop();
    helper(i+1,nums,temp,res);
};

File: C++/78. Subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> res;
        vector<int> temp;
        helper(0,nums,temp,res);
        return res;
    }
    void helper(int i,vector<int> &nums,vector<int> &temp,vector<vector<int>> &res){
        if(i==nums.size()){
            res.push_back(temp);
            return;
        }
        temp.push_back(nums[i]);
        helper(i+1,nums,temp,res);
        temp.pop_back();
        helper(i+1,nums,temp,res);
    }
};
